src/log/log.c

```diff
@@ -2117,7 +2131,20 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
             lastlog = WT_MAX(lastlog, lognum);
             firstlog = WT_MIN(firstlog, lognum);
         }
-        WT_SET_LSN(&start_lsn, firstlog, 0);
+        if (start_range_lsnp != NULL) {
```

Comment:
Can you elaborate on why this code block appears twice?

Reply:
So that the start_range is considered in both the cases of either having a log or not

Reply:
I think you could move all of the start_range checking to above line 2151 and delete the duplicated block. Additionally you can can leave the `WT_SET_LSN(&start_lsn, firstlog, 0);` in that scope without any if else wrapping it. That way it'll get set and overwritten if `start_range_lsnp` isn't NULL.

Reply:
Yeah this makes sense